fixing bugs from changing on my files

AhmedHessuin opened the pull request:
adding first_fix 
adding more methods on Void and memory

AhmedHessuin committed:
fixing bugs from changing on my files 
adding first_fix 
adding more methods on Void and memory

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Void.java:
```diff
@@ -21,55 +28,83 @@ we have intialy void (free locations)= base = 0 limit 1000
-    private Vector<Segment> void_vector;
 
+    private Vector<Segment> void_vector;
+    private char condtion = 'b';// b = sort on base,  h=sort on larger at top , 
```
No initialize for variable here. Initialization in constructors unless static variables.

AhmedHessuin replied:
fixed

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Void.java:
```diff
@@ -21,55 +28,83 @@ we have intialy void (free locations)= base = 0 limit 1000
+ * @author ahmed
  */
 public class Void {
+//   Collections.sort(xx,(a, b) -> a.getBase() < b.getBase() ? -1 : a.getBase() == b.getBase() ? 0 : 1);
```
this comment is irrelevant here to this section. it must be removed or moved.

AhmedHessuin replied:
fixed

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Segment.java:
```diff
@@ -8,8 +17,18 @@
-
 // ==============constructors ==========================//
+
+    //use this for gui holes
```
will be never used. because it give an ID to hole and it's wrong.it must be removed or use a constructor parameter boolean isPartofProcess as I did.

AhmedHessuin replied:
fixed

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Segment.java:
```diff
@@ -8,8 +17,18 @@
     private int base;
     private int limit;
     private boolean inserted = false;
```
no initialization here.

AhmedHessuin replied:
fixed

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Segment.java:
```diff
@@ -1,5 +1,14 @@
+ */
 public class Segment {
 
     private static int SEGMENT_ID = 0;
```
there must be a way to reset the ID every process. So the segments of the process always start with ID 0.

AhmedHessuin replied:
not fixed
every segment got it elite  id 
not need to give it a zero .

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Process.java:
```diff
@@ -145,14 +197,19 @@ public void remove_Segment_i(int i) {
 
+    public void print(int i) {
+
+        getSegment_vector().get(i).print();
```
Don't use functions to get class parameters in the class. to minimize stack overhead.

AhmedHessuin replied:
need to print have another solution say it

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Process.java:
```diff
@@ -1,140 +1,192 @@
     private int ID;
-    private Color color;
     private boolean inserted = false;
+    private int number_of_segments;
```
this is a stupid redundant. every vector has its own size and its complicity is O(1). So you recount after him!

AhmedHessuin replied:
this was a basic algorthim (naive) need to be stupid .
you are right fixed.

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Memory.java:
```diff
@@ -55,37 +68,40 @@ public int get_total_free() {
     }
 
-    public void swap_base_2_segments(Segment first, Segment second) {
+    private void swap_base_2_segments(Segment first, Segment second) {
```
This is a wrong function implementation or wrong named.
suppose 2 segments that don't lay after each other.

AhmedHessuin replied:
it's for sorting just to swap the main function handle the logic .
not fixed

IbrahimAmrIbrahim reviewed (changes requested):
you need to change the things in my reviews to merge your changes.

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Memory.java:
```diff
@@ -1,47 +1,60 @@
 package memorymanagementAlgorithm;
 
 import java.util.*;
+import javafx.scene.paint.Color;
```
this import never used. please remove it.

AhmedHessuin replied:
gonna fix it, fair enough.

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Process.java:
```diff
@@ -1,140 +1,192 @@
     //====================constructor============================//
-    public Process() {
+
+    public Process()// just the id
```
never put a comment like this between function name and its start bracket. So the auto-formatting can committed.

AhmedHessuin replied:
in c++ this show the comment on when you use the function but it dosen't work here gonna fix it ./
not fixed .

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Process.java:
```diff
@@ -1,140 +1,192 @@
         Process_ID++;
+        Segment_vector = new Vector<Segment>();
         Segment_vector = input;
+        number_of_segments = Segment_vector.size();
```
this is a stupid redundant. every vector has its own size and its complicity is O(1). So you recount after him!

AhmedHessuin replied:
same answer naive algorthim first then advanced.

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Process.java:
```diff
@@ -1,140 +1,192 @@
-        Segment_vector.add(input);
-        Collections.sort(Segment_vector, (a, b) -> a.getLimit() < b.getLimit() ? -1 : a.getLimit() == b.getLimit() ? 0 : 1);
+        getSegment_vector().add(input);
+        number_of_segments = getSegment_vector().size();
```
this is a stupid redundant. every vector has its own size and its complicity is O(1). So you recount after him!

AhmedHessuin replied:
naive

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Process.java:
```diff
@@ -1,140 +1,192 @@
+    {
         ID = Process_ID;
         Process_ID++;
         Segment_vector = new Vector<Segment>();
```
No initialize here.

AhmedHessuin replied:
where is is the initalize ?! maybe the comment and format.

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Process.java:
```diff
@@ -1,140 +1,192 @@
         Process_ID++;
         Segment_vector = new Vector<Segment>();
         Segment_vector.add(input);
+        number_of_segments = Segment_vector.size();
```
this is a stupid redundant. every vector has its own size and its complicity is O(1). So you recount after him!

AhmedHessuin replied:
naive

IbrahimAmrIbrahim commented on MemoryManagement/src/memorymanagementAlgorithm/Process.java:
```diff
@@ -1,140 +1,192 @@
-        Segment_vector.addAll(input);
-        Collections.sort(Segment_vector, (a, b) -> a.getLimit() < b.getLimit() ? -1 : a.getLimit() == b.getLimit() ? 0 : 1);
+        getSegment_vector().addAll(input);
+         number_of_segments = getSegment_vector().size();
```
this is a stupid redundant. every vector has its own size and its complicity is O(1). So you recount after him!

AhmedHessuin replied:
naive

AhmedHessuin committed:
attributes long 
adding method in memory add_old process
changing in memory attributes  and constructor
add one more constructor for in Segment for free and old (see this for gui

AhmedHessuin committed:
fixing constructor issue 
fixing problem with allocated_segment and the process_runing 
where the segment didn't hit an affect on the runing process

AhmedHessuin committed:
fixed comment issue

AhmedHessuin committed:
removed a comment
